machine/v1beta1/types_alibabaprovider.go

```diff
@@ -0,0 +1,398 @@
+	// https://www.alibabacloud.com/help/doc-detail/101348.htm?spm=a2c63.p38356.879954.48.78f0199aX3dfIE
+	SecurityGroups []AlibabaResourceReference `json:"securityGroups,omitempty"`
+
+	// Bandwidth describes the internet bandwidth strategy for the instance
+	// +optional
+	Bandwidth BandwidthProperties `json:"bandwidth,omitempty"`
+
+	// Information regarding the the system disk for the instance
+	// +optional
+	SystemDisk SystemDiskProperties `json:"systemDisk,omitempty"`
+
+	// VSwitch is a reference to the vswitch to use for this instance
+	// This parameter is required when you create an instance of the VPC type.
+	// You can call the DescribeVSwitches operation to query the created vSwitches.
+	VSwitch AlibabaResourceReference `json:"vSwitch,omitempty"`
+
+	// Subscription holds the billing information for the instance
+	// +optional
+	Subscription SubscriptionInfo `json:"subscription,omitempty"`
+
+	// Description of the instance. The description must be 2 to 256 characters in length and cannot start with http:// or https://.
+	// This parameter is empty by default.
+	// +optional
+	Description string `json:"description,omitempty"`
+
+	//The ID of the vSwitch to which to connect the instance. This parameter is required when you create an instance of the VPC type. You can call the DescribeVSwitches operation to query the created vSwitches.
+	// +optional
+	VSwitchID string `json:"vSwitchId,omitempty"`
+
+	//The billing method of the instance. Default value: PostPaid. Valid values:
+	//
+	//PrePaid: subscription. If you set this parameter to PrePaid, make sure that you have sufficient balance or credit within your account. Otherwise, an InvalidPayMethod error is returned.
+	//PostPaid: pay-as-you-go.
+	// +optional
+	InstanceChargeType string `json:"instanceChargeType,omitempty"`
+
+	//The name of the instance Resource Access Management (RAM) role. You can call the ListRoles operation provided by RAM to query the instance RAM roles that you have created.
+	// +optional
+	RAMRoleName string `json:"ramRoleName,omitempty"`
+
+	// SecurityEnhancementStrategy specifies whether to enable security hardening.
+	// Valid values:
+	// Active: enables security hardening. This value is applicable only to public images.
+	// Deactive: does not enable security hardening. This value is applicable to all images.
+	// +optional
+	SecurityEnhancementStrategy string `json:"securityEnhancementStrategy,omitempty"`
+
+	// ResourceGroupID is the unique ID of the resource group to which to assign the instance.
+	// +optional
+	ResourceGroupID string `json:"resourceGroupId,omitempty"`
+
+	// DeletionProtection is release protection property of the instance. It specifies whether you can use the ECS console or call the DeleteInstance operation to manually release the instance. Default value: false. Valid values:
+	// true: enables release protection.
+	// false: disables release protection.
+	// +optional
+	DeletionProtection bool `json:"deletionProtection,omitempty"`
```

Comment:
Bools are typically not recommended for APIs, would it make sense to make this an enum instead?

Reply:
Will make an attempt at an enum.

Reply:
Fixed